lib/edgehog_device_forwarder/connections.ex

```diff
@@ -0,0 +1,70 @@
+# Copyright 2023 SECO Mind Srl
+# SPDX-License-Identifier: Apache-2.0
+
+defmodule EdgehogDeviceForwarder.Connections do
+  use Supervisor
+
+  alias EdgehogDeviceForwarder.Connections.TokenData
+
+  @tokens Application.compile_env!(:edgehog_device_forwarder, EdgehogDeviceForwarder.Connections)[
+            :tokens
+          ]
+
+  @type token :: String.t()
+  @type device_id :: TokenData.device_id()
+  @type realm :: TokenData.realm()
+  @type socket :: TokenData.socket()
+
+  # API
+
+  ## Token
+  @spec register_device_info(token, %{:device_id => device_id, :realm => realm}) ::
+          :ok | {:error, :token_not_found}
+  def register_device_info(token, device_info) do
+    ConCache.update(@tokens, token, &TokenData.put_device_info(&1, device_info))
+  end
+
+  @spec register_socket(token, socket) :: :ok | {:error, :already_exists}
+  def register_socket(token, socket) do
+    # TODO: check if we need to handle socket's process termination ourself.
+    # I'd be inclined to say no from
+    #   - https://hexdocs.pm/phoenix/Phoenix.Socket.Transport.html#c:terminate/2
+    #   - https://hexdocs.pm/phoenix/Phoenix.Socket.Transport.html#module-custom-transports
+    ConCache.insert_new(@tokens, token, TokenData.new(socket))
```

Comment:
Can you expand on the comment above? Because my impression is that the cache is going to grow indefinitely because if a socket terminates, it won't clean up its entries here.

Reply:
Yes. 
As you said, in [`GenServer.terminate/2`](https://hexdocs.pm/elixir/1.15.7/GenServer.html#c:terminate/2) there is the sentence "However, it is not guaranteed that `terminate/2` is called when a `GenServer` exits.".

However, in the Phoenix.Socket.Transport documentation (linked in the comment) it looks to me as if it is implied that the socket will always call `terminate/2` before terminating.

If that is the case, it is safe to just drop the token from the socket itself on termination (and so the cache won't grow indefinitely).

Reply:
Implemented in #15